\documentclass[a4paper, twoside]{article}%
%Options -- Point size:  10pt (default), 11pt, 12pt
%        -- Paper size:  letterpaper (default), a4paper, a5paper, b5paper
%                        legalpaper, executivepaper
%        -- Orientation  (portrait is the default)
%                        landscape
%        -- Print size:  oneside (default), twoside
%        -- Quality      final(default), draft
%        -- Title page   notitlepage, titlepage(default)
%        -- Columns      onecolumn(default), twocolumn
%        -- Equation numbering (equation numbers on the right is the default)
%                        leqno
%        -- Displayed equations (centered is the default)
%                        fleqn (equations start at the same distance from the right side)
%        -- Open bibliography style (closed is the default)
%                        openbib
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Lista de Paquetes
\usepackage{pifont}             % Maneja Fuentes 
%\usepackage{PTSans}            % Maneja Fuentes Sans Serif
%\usepackage{skak}              % simbolos de ajedrez
\usepackage[T1]{fontenc}        % Texto de alta calidad en pdfs
\usepackage{ae,aecompl}         % Texto de alta calidad en pdfs
\usepackage{pslatex}            % Texto de alta calidad en pdfs
\usepackage{textcomp}           % Cosas de Texto
\usepackage{multicol}           % Para manejar multiples columnas
\usepackage[ansinew]{inputenc}  % Para poner acentos
\usepackage[spanish,es-tabla]{babel}     % Para separar palabras al fin de los
% renglones.
\usepackage{amsmath}            % Para hacer matem√°tica
\usepackage{amsfonts}           % Para hacer matem√°tica
\usepackage{amssymb}            % Para hacer matem√°tica
\usepackage{enumerate}          % Para Manehar Enumeraciones
\usepackage{vwcol}              % para manejar multiples columas
\usepackage{graphicx}           % Para manejar los gr√°ficos
\usepackage{fancyhdr}           % Para manejar encabezados y pie de p√°ginas
\usepackage{lastpage}           % Crea una referencia a la √∫ltima p√°gina. Util
                                % para numerar p√°ginas de la forma "p√°gina x de y"
%\usepackage[final]{ps4pdf}     % para incluir archivos eps dentro del documento pdf
\usepackage{array}
\usepackage{soul}
\usepackage{boolexpr}
\usepackage[hyphens]{url}   % Typeset urls
\usepackage[pdfborder={0 0 0 [3 3]}]{hyperref}       % HipervÌsnculos en las url
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% ConfiguraciÛn de p·gina
\textwidth 15cm               
\textheight 25.7cm            
\oddsidemargin 1cm           
\evensidemargin 0cm          
\topmargin -1.5cm             
\headheight 0cm               
\headsep 10pt                 
\marginparwidth 0cm
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Bibliograf√≠a
\bibliographystyle{alpha}      
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Titulo
\newcommand{\s}{\the\textwidth}
\newcommand{\materiaName}{Bases de datos genomicas}
\newcommand{\seminarioName}{Guia de trabajos practicos}
\newcommand{\fecha}{Julio de 2015}
\newcommand{\titulo}{
  \begin{center}
  \vspace*{-22pt}\large\textbf{\materiaName}\\
  \vspace{2pt}\normalsize{\seminarioName}\\
  \vspace{2pt}\small{MaestrÌa en bioinform·tica y biologÌa de sistemas}\\
  \vspace{-6pt}\rule{\textwidth}{0.4pt}
  \end{center}
}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Encabezado y Pie de P√°gina
 \headheight=14pt
 \renewcommand{\headrulewidth}{0.4pt}                                 
 \renewcommand{\footrulewidth}{0.4pt}                                 
 \pagestyle{fancy}                                                   
 \fancyfoot{}
 \lfoot{\footnotesize{P·gina \thepage\ de \pageref{LastPage}}}        
 \lhead{\footnotesize{\materiaName}}
 \rhead{\footnotesize{\fecha\  - \seminarioName}}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Fuentes
%\renewcommand{\familydefault}{\sfdefault} % Cambia todas las fuentes a sans 
                                           % Serif
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
                                 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Enumeraciones
\renewcommand{\theenumi}{\alph{enumi}}
\renewcommand{\labelenumi}{\textbf{\theenumi}}
\renewcommand{\theenumii}{\arabic{enumii}}
\renewcommand{\labelenumii}{\theenumii}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Fin del preludio
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}
\titulo
 
\section{Bases de datos primarias de nucleÛtidos}

\subsection{National Center of Biotechnology Information - NCBI}
El NCBI ofrece un conjunto de herramientos para el acceso program·tico a sus 
bases de datos que se llama NCBI E-utilities y se basa en una interface REST.
EInfo
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi
Functions

    Provides a list of the names of all valid Entrez databases
    Provides statistics for a single database, including lists of indexing fields and available link names

Required Parameters

None. If no db parameter is provided, einfo will return a list of the names of all valid Entrez databases.
Optional Parameters
db

Target database about which to gather statistics. Value must be a valid Entrez database name.
version

Used to specify version 2.0 EInfo XML. The only supported value is `2.0'. When present, EInfo will return XML that includes two new fields: <IsTruncatable> and <IsRangeable>. Fields that are truncatable allow the wildcard character `*' in
terms. The wildcard character will expand to match any set of characters up to a limit of 600 unique expansions. Fields that are rangeable allow the range operator `:' to be placed between a lower and upper limit for the desired range (e.g.
2008:2010[pdat]).
retmode

Retrieval type. Determines the format of the returned output. The default value is `xml' for EInfo XML, but `json' is also supported to return output in JSON format.
Examples

Return a list of all Entrez database names:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi

Return version 2.0 statistics for Entrez Protein:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi?db=protein&version=2.0
Go to:
ESearch
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi
Functions

    Provides a list of UIDs matching a text query
    Posts the results of a search on the History server
    Downloads all UIDs from a dataset stored on the History server
    Combines or limits UID datasets stored on the History server
    Sorts sets of UIDs

Required Parameters
db

Database to search. Value must be a valid Entrez database name (default = pubmed).
term

Entrez text query. All special characters must be URL encoded. Spaces may be replaced by '+' signs. For very long queries (more than several hundred characters long), consider using an HTTP POST call. See the PubMed or Entrez help for information about search field descriptions and tags. Search fields and tags are database specific.

esearch.fcgi?db=pubmed&term=asthma

Optional Parameters - History Server
usehistory

When usehistory is set to 'y', ESearch will post the UIDs resulting from the search operation onto the History server so that they can be used directly in a subsequent E-utility call. Also, usehistory must be set to 'y' for ESearch to interpret query key values included in term or to accept a WebEnv as input.
WebEnv

Web environment string returned from a previous ESearch, EPost or ELink call. When provided, ESearch will post the results of the search operation to this pre-existing WebEnv, thereby appending the results to the existing environment. In addition, providing WebEnv allows query keys to be used in term so that previous search sets can be combined or limited. As described above, if WebEnv is used, usehistory must be set to 'y'.

esearch.fcgi?db=pubmed&term=asthma&WebEnv=<webenv string>&usehistory=y

query_key

Integer query key returned by a previous ESearch, EPost or ELink call. When provided, ESearch will find the intersection of the set specified by query_key and the set retrieved by the query in term (i.e. joins the two with AND). For query_key to function, WebEnv must be assigned an existing WebEnv string and usehistory must be set to 'y'.

Values for query keys may also be provided in term if they are preceeded by a '#' (%23 in the URL). While only one query_key parameter can be provided to ESearch, any number of query keys can be combined in term. Also, if query keys are provided in term, they can be combined with OR or NOT in addition to AND.

The following two URLs are functionally equivalent:

esearch.fcgi?db=pubmed&term=asthma&query_key=1&WebEnv=
<webenv string>&usehistory=y

esearch.fcgi?db=pubmed&term=%231+AND+asthma&WebEnv=
<webenv string>&usehistory=y

Optional Parameters - Retrieval
retstart

Sequential index of the first UID in the retrieved set to be shown in the XML output (default=0, corresponding to the first record of the entire set). This parameter can be used in conjunction with retmax to download an arbitrary subset of UIDs retrieved from a search.
retmax

Total number of UIDs from the retrieved set to be shown in the XML output (default=20). By default, ESearch only includes the first 20 UIDs retrieved in the XML output. If usehistory is set to 'y', the remainder of the retrieved set will be stored on the History server; otherwise these UIDs are lost. Increasing retmax allows more of the retrieved UIDs to be included in the XML output, up to a maximum of 100,000 records. To retrieve more than 100,000 UIDs, submit multiple esearch requests while incrementing the value of retstart (see Application 3).
rettype

Retrieval type. There are two allowed values for ESearch: 'uilist' (default), which displays the standard XML output, and 'count', which displays only the <Count> tag.
retmode

Retrieval type. Determines the format of the returned output. The default value is `xml' for ESearch XML, but `json' is also supported to return output in JSON format.
sort

Specifies the method used to sort UIDs in the ESearch output. The available values vary by database (db) and may be found in the Display Settings menu on an Entrez search results page. If usehistory is set to `y', the UIDs are loaded onto the History Server in the specified sort order and will be retrieved in that order by ESummary or EFetch. Example values are `relevance' and `name' for Gene and `first+author' and `pub+date' for PubMed. Users should be aware that the default value of sort varies from one database to another, and that the default value used by ESearch for a given database may differ from that used on NCBI web search pages.
field

Search field. If used, the entire search term will be limited to the specified Entrez field. The following two URLs are equivalent:

esearch.fcgi?db=pubmed&term=asthma&field=title

esearch.fcgi?db=pubmed&term=asthma[title]

Optional Parameters - Dates
datetype

Type of date used to limit a search. The allowed values vary between Entrez databases, but common values are 'mdat' (modification date), 'pdat' (publication date) and 'edat' (Entrez date). Generally an Entrez database will have only two allowed values for datetype.
reldate

When reldate is set to an integer n, the search returns only those items that have a date specified by datetype within the last n days.
mindate, maxdate

Date range used to limit a search result by the date specified by datetype. These two parameters (mindate, maxdate) must be used together to specify an arbitrary date range. The general date format is YYYY/MM/DD, and these variants are also allowed: YYYY, YYYY/MM.
Examples

Search in PubMed with the term cancer for abstracts that have an Entrez date within the last 60 days; retrieve the first 100 PMIDs and translations; post the results on the History server and return a WebEnv and query_key:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&term=cancer&reldate=60&datetype=edat&retmax=100&usehistory=y

Search in PubMed for the journal PNAS, Volume 97, and retrieve six PMIDs starting with the seventh PMID in the list:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&term=PNAS[ta]+AND+97[vi]&retstart=6&retmax=6&tool=biomed3

Search in the NLM Catalog for journals matching the term obstetrics:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=nlmcatalog&term=obstetrics+AND+ncbijournals[filter]

Search PubMed Central for free full text articles containing the query stem cells:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pmc&term=stem+cells+AND+free+fulltext[filter]

Search in Nucleotide for all tRNAs:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=nucleotide&term=biomol+trna[prop]

Search in Protein for a molecular weight range:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=protein&term=70000:90000[molecular+weight]
Go to:
EPost
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/epost.fcgi
Functions

    Uploads a list of UIDs to the Entrez History server
    Appends a list of UIDs to an existing set of UID lists attached to a Web Environment

Required Parameters
db

Database containing the UIDs in the input list. The value must be a valid Entrez database name (default = pubmed).
id

UID list. Either a single UID or a comma-delimited list of UIDs may be provided. All of the UIDs must be from the database specified by db. There is no set maximum for the number of UIDs that can be passed to epost, but if more than about 200 UIDs are to be posted, the request should be made using the HTTP POST method.

epost.fcgi?db=protein&id=15718680,157427902,119703751

Optional Parameter
WebEnv

Web Environment. If provided, this parameter specifies the Web Environment that will receive the UID list sent by post. EPost will create a new query key associated with that Web Environment. Usually this WebEnv value is obtained from the output of a previous ESearch, EPost or ELink call. If no WebEnv parameter is provided, EPost will create a new Web Environment and post the UID list to query_key 1.

epost.fcgi?db=protein&id=15718680,157427902,119703751&WebEnv=
<webenv string>

Example

Post records to PubMed:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/epost.fcgi?db=pubmed&id=11237011,12466850
Go to:
ESummary
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi
Functions

    Returns document summaries (DocSums) for a list of input UIDs
    Returns DocSums for a set of UIDs stored on the Entrez History server

Required Parameter
db

Database from which to retrieve DocSums. The value must be a valid Entrez database name (default = pubmed).
Required Parameter - Used only when input is from a UID list
id

UID list. Either a single UID or a comma-delimited list of UIDs may be provided. All of the UIDs must be from the database specified by db. There is no set maximum for the number of UIDs that can be passed to ESummary, but if more than about 200 UIDs are to be provided, the request should be made using the HTTP POST method.

esummary.fcgi?db=protein&id=15718680,157427902,119703751

Required Parameters - Used only when input is from the Entrez History server
query_key

Query key. This integer specifies which of the UID lists attached to the given Web Environment will be used as input to ESummary. Query keys are obtained from the output of previous ESearch, EPost or ELink calls. The query_key parameter must be used in conjunction with WebEnv.
WebEnv

Web Environment. This parameter specifies the Web Environment that contains the UID list to be provided as input to ESummary. Usually this WebEnv value is obtained from the output of a previous ESearch, EPost or ELink call. The WebEnv parameter must be used in conjunction with query_key.

esummary.fcgi?db=protein&query_key=<key>&WebEnv=<webenv string>

Optional Parameters - Retrieval
retstart

Sequential index of the first DocSum to be retrieved (default=1, corresponding to the first record of the entire set). This parameter can be used in conjunction with retmax to download an arbitrary subset of DocSums from the input set.
retmax

Total number of DocSums from the input set to be retrieved, up to a maximum of 10,000. If the total set is larger than this maximum, the value of retstart can be iterated while holding retmax constant, thereby downloading the entire set in batches of size retmax.
retmode

Retrieval type. Determines the format of the returned output. The default value is `xml' for ESummary XML, but `json' is also supported to return output in JSON format.
version

Used to specify version 2.0 ESummary XML. The only supported value is `2.0'. When present, ESummary will return version 2.0 DocSum XML that is unique to each Entrez database and that often contains more data than the default DocSum XML.
Examples

PubMed:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id=11850928,11482001

PubMed, version 2.0 XML:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id=11850928,11482001&version=2.0

Protein:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=protein&id=28800982,28628843

Nucleotide:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=nucleotide&id=28864546,28800981

Structure:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=structure&id=19923,12120

Taxonomy:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=taxonomy&id=9913,30521

UniSTS:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=unists&id=254085,254086
Go to:
EFetch
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi
Functions

    Returns formatted data records for a list of input UIDs
    Returns formatted data records for a set of UIDs stored on the Entrez History server

Required Parameters
db

Database from which to retrieve records. The value must be a valid Entrez database name (default = pubmed). Currently EFetch does not support all Entrez databases. Please see Table 1 in Chapter 2 for a list of available databases.
Required Parameter - Used only when input is from a UID list
id

UID list. Either a single UID or a comma-delimited list of UIDs may be provided. All of the UIDs must be from the database specified by db. There is no set maximum for the number of UIDs that can be passed to EFetch, but if more than about 200 UIDs are to be provided, the request should be made using the HTTP POST method.

efetch.fcgi?db=protein&id=15718680,157427902,119703751

Required Parameters - Used only when input is from the Entrez History server
query_key

Query key. This integer specifies which of the UID lists attached to the given Web Environment will be used as input to EFetch. Query keys are obtained from the output of previous ESearch, EPost or ELInk calls. The query_key parameter must be used in conjunction with WebEnv.
WebEnv

Web Environment. This parameter specifies the Web Environment that contains the UID list to be provided as input to EFetch. Usually this WebEnv value is obtained from the output of a previous ESearch, EPost or ELink call. The WebEnv parameter must be used in conjunction with query_key.

efetch.fcgi?db=protein&query_key=<key>&WebEnv=<webenv string>

Optional Parameters - Retrieval
retmode

Retrieval mode. This parameter specifies the data format of the records returned, such as plain text, HMTL or XML. See Table 1 for a full list of allowed values for each database.
Table 1
Table 1

- Valid values of &retmode and &rettype for EFetch (null = empty string)
rettype

Retrieval type. This parameter specifies the record view returned, such as Abstract or MEDLINE from PubMed, or GenPept or FASTA from protein. Please see Table 1 for a full list of allowed values for each database.
retstart

Sequential index of the first record to be retrieved (default=0, corresponding to the first record of the entire set). This parameter can be used in conjunction with retmax to download an arbitrary subset of records from the input set.
retmax

Total number of records from the input set to be retrieved, up to a maximum of 10,000. Optionally, for a large set the value of retstart can be iterated while holding retmax constant, thereby downloading the entire set in batches of size retmax.
Optional Parameters - Sequence Databases
strand

Strand of DNA to retrieve. Available values are "1" for the plus strand and "2" for the minus strand.
seq_start

First sequence base to retrieve. The value should be the integer coordinate of the first desired base, with "1" representing the first base of the seqence.
seq_stop

Last sequence base to retrieve. The value should be the integer coordinate of the last desired base, with "1" representing the first base of the seqence.
complexity

Data content to return. Many sequence records are part of a larger data structure or "blob", and the complexity parameter determines how much of that blob to return. For example, an mRNA may be stored together with its protein product. The available values are as follows:
Value of complexity Data returned for each requested GI
0 entire blob
1 bioseq
2 minimal bioseq-set
3 minimal nuc-prot
4 minimal pub-set
Examples

PubMed

Fetch PMIDs 17284678 and 9997 as text abstracts:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=17284678,9997&retmode=text&rettype=abstract

Fetch PMIDs in XML:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&id=11748933,11700088&retmode=xml

PubMed Central

Fetch XML for PubMed Central ID 212403:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pmc&id=212403

Nucleotide/Nuccore

Fetch the first 100 bases of the plus strand of GI 21614549 in FASTA format:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=21614549&strand=1&seq_start=1&seq_stop=100&rettype=fasta&retmode=text

Fetch the first 100 bases of the minus strand of GI 21614549 in FASTA format:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=21614549&strand=2&seq_start=1&seq_stop=100&rettype=fasta&retmode=text

Fetch the nuc-prot object for GI 21614549:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=21614549&complexity=3

Fetch the full ASN.1 record for GI 5:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=5

Fetch FASTA for GI 5:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=5&rettype=fasta

Fetch the GenBank flat file for GI 5:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=5&rettype=gb

Fetch GBSeqXML for GI 5:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=5&rettype=gb&retmode=xml

Fetch TinySeqXML for GI 5:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=5&rettype=fasta&retmode=xml

Popset

Fetch the GenPept flat file for Popset ID 12829836:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=popset&id=12829836&rettype=gp

Protein

Fetch the GenPept flat file for GI 8:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=protein&id=8&rettype=gp

Fetch GBSeqXML for GI 8:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=protein&id=8&rettype=gp&retmode=xml

Sequences

Fetch FASTA for a transcript and its protein product (GIs 312836839 and 34577063)

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=sequences&id=312836839,34577063&rettype=fasta&retmode=text

Gene

Fetch full XML record for Gene ID 2:

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=gene&id=2&retmode=xml
Go to:
ELink
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi
Functions

    Returns UIDs linked to an input set of UIDs in either the same or a different Entrez database
    Returns UIDs linked to other UIDs in the same Entrez database that match an Entrez query
    Checks for the existence of Entrez links for a set of UIDs within the same database
    Lists the available links for a UID
    Lists LinkOut URLs and attributes for a set of UIDs
    Lists hyperlinks to primary LinkOut providers for a set of UIDs
    Creates hyperlinks to the primary LinkOut provider for a single UID

Required Parameters
db

Database from which to retrieve UIDs. The value must be a valid Entrez database name (default = pubmed). This is the destination database for the link operation.
dbfrom

Database containing the input UIDs. The value must be a valid Entrez database name (default = pubmed). This is the origin database of the link operation. If db and dbfrom are set to the same database value, then ELink will return computational neighbors within that database. Please see the full list of Entrez links for available computational neighbors. Computational neighbors have linknames that begin with dbname_dbname (examples: protein_protein, pcassay_pcassay_activityneighbor).
cmd

ELink command mode. The command mode specified which function ELink will perform. Some optional parameters only function for certain values of &cmd (see below).

cmd=neighbor (default)

ELink returns a set of UIDs in db linked to the input UIDs in dbfrom.

Example: Link from protein to gene

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&db=gene&id=15718680,157427902

cmd=neighbor_score

ELink returns a set of UIDs within the same database as the input UIDs along with computed similarity scores.

Example: Find related articles to PMID 20210808

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&db=pubmed&id=20210808&cmd=neighbor_score

cmd=neighbor_history

ELink posts the output UIDs to the Entrez History server and returns a query_key and WebEnv corresponding to the location of the output set.

Example: Link from protein to gene and post the results on the Entrez History

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&db=gene&id=15718680,157427902&cmd=neighbor_history

cmd=acheck

ELink lists all links available for a set of UIDs.

Example: List all possible links from two protein GIs

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&id=15718680,157427902&cmd=acheck

Example: List all possible links from two protein GIs to PubMed

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&db=pubmed&id=15718680,157427902&cmd=acheck

cmd=ncheck

ELink checks for the existence of links within the same database for a set of UIDs. These links are equivalent to setting db and dbfrom to the same value.

Example: Check whether two nuccore sequences have "related sequences" links.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=nuccore&id=21614549,219152114&cmd=ncheck

cmd=lcheck

Elink checks for the existence of external links (LinkOuts) for a set of UIDs.

Example: Check whether two protein sequences have any LinkOut providers.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&id=15718680,157427902&cmd=lcheck

cmd=llinks

For each input UID, ELink lists the URLs and attributes for the LinkOut providers that are not libraries.

Example: List the LinkOut URLs for non-library providers for two pubmed abstracts.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=19880848,19822630&cmd=llinks

cmd=llinkslib

For each input UID, ELink lists the URLs and attributes for all LinkOut providers including libraries.

Example: List all LinkOut URLs for two PubMed abstracts.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=19880848,19822630&cmd=llinkslib

cmd=prlinks

ELink lists the primary LinkOut provider for each input UID, or links directly to the LinkOut provider's web site for a single UID if retmode is set to ref.

Example: Find links to full text providers for two PubMed abstracts.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=19880848,19822630&cmd=prlinks

Example: Link directly to the full text for a PubMed abstract at the provider's web site.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=19880848&cmd=prlinks&retmode=ref
Required Parameter - Used only when input is from a UID list
id

UID list. Either a single UID or a comma-delimited list of UIDs may be provided. All of the UIDs must be from the database specified by dbfrom. There is no set maximum for the number of UIDs that can be passed to ELink, but if more than about 200 UIDs are to be provided, the request should be made using the HTTP POST method.

Link from protein to gene.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&db=gene&id=15718680,157427902,119703751

Find related sequences (link from nuccore to nuccore).

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=nuccore&db=nuccore&id=34577062

If more than one id parameter is provided, ELink will perform a separate link operation for the set of UIDs specified by each id parameter. This effectively accomplishes "one-to-one" links and preserves the connection between the input and output UIDs.

Find one-to-one links from protein to gene.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=protein&db=gene&id=15718680&id=157427902&id=119703751
Required Parameters - Used only when input is from the Entrez History server
query_key

Query key. This integer specifies which of the UID lists attached to the given Web Environment will be used as input to ELink. Query keys are obtained from the output of previous ESearch, EPost or ELInk calls. The query_key parameter must be used in conjunction with WebEnv.
WebEnv

Web Environment. This parameter specifies the Web Environment that contains the UID list to be provided as input to ELink. Usually this WebEnv value is obtained from the output of a previous ESearch, EPost or ELink call. The WebEnv parameter must be used in conjunction with query_key.

Link from protein to gene:
elink.fcgi?dbfrom=protein&db=gene&query_key=<key>&WebEnv=<webenv string>

Find related sequences (link from protein to protein):
elink.fcgi?dbfrom=protein&db=protein&query_key=<key>&WebEnv=
<webenv string>

Optional Parameters - Limiting the Output Set of Links
linkname

Name of the Entrez link to retrieve. Every link in Entrez is given a name of the form

dbfrom_db_subset.

The values of subset vary depending on the values of dbfrom and db. Many dbfrom/db combinations have no subset values. See the list of Entrez links for a listing of all available linknames. When linkname is used, only the links with that name will be retrieved.

The linkname parameter only functions when cmd is set to neighbor or neighbor_history.

Find all links from gene to snp.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=gene&db=snp&id=93986

Find snps with genotype data linked to genes.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=gene&db=snp&id=93986&linkname=gene_snp_genegenotype
term

Entrez query used to limit the output set of linked UIDs. The query in the term parameter will be applied after the link operation, and only those UIDs matching the query will be returned by ELink. The term parameter only functions when db and dbfrom are set to the same database value.

Find all related articles for a PMID.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&db=pubmed&id=19879512

Find all related review articles published in 2008 for a PMID.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&db=pubmed&id=19879512&term=review%5Bfilter%5D+AND+2008%5Bpdat%5Dh
holding

Name of LinkOut provider. Only URLs for the LinkOut provider specified by holding will be returned. The value provided to holding should be the abbreviation of the LinkOut provider's name found in the <NameAbbr> tag of the ELink XML output when cmd is set to llinks or llinkslib. The holding parameter only functions when cmd is set to llinks or llinkslib.

Find information for all LinkOut providers for a PMID.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&cmd=llinkslib&id=16210666

Find information from clinicaltrials.gov for a PMID.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&cmd=llinkslib&id=16210666&holding=CTgov
Optional Parameters - Dates

These parameters only function when cmd is set to neighbor or neighbor_history and dbfrom is pubmed.
datetype

Type of date used to limit a link operation. The allowed values vary between Entrez databases, but common values are 'mdat' (modification date), 'pdat' (publication date) and 'edat' (Entrez date). Generally an Entrez database will have only two allowed values for datetype.
reldate

When reldate is set to an integer n, ELink returns only those items that have a date specified by datetype within the last n days.
mindate, maxdate

Date range used to limit a link operation by the date specified by datetype. These two parameters (mindate, maxdate) must be used together to specify an arbitrary date range. The general date format is YYYY/MM/DD, and these variants are also allowed: YYYY, YYYY/MM.
Go to:
EGQuery
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/egquery.fcgi
Function

Provides the number of records retrieved in all Entrez databases by a single text query.
Required Parameter
term

Entrez text query. All special characters must be URL encoded. Spaces may be replaced by '+' signs. For very long queries (more than several hundred characters long), consider using an HTTP POST call. See the PubMed or Entrez help for information about search field descriptions and tags. Search fields and tags are database specific.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/egquery.fcgi?term=asthma
Go to:
ESpell
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/espell.fcgi
Function

Provides spelling suggestions for terms within a single text query in a given database.
Required Parameters
db

Database to search. Value must be a valid Entrez database name (default = pubmed).
term

Entrez text query. All special characters must be URL encoded. Spaces may be replaced by '+' signs. For very long queries (more than several hundred characters long), consider using an HTTP POST call. See the PubMed or Entrez help for information about search field descriptions and tags. Search fields and tags are database specific.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/espell.fcgi?db=pubmed&term=asthmaa+OR+alergies
Go to:
ECitMatch
Base URL

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/ecitmatch.cgi
Function

Retrieves PubMed IDs (PMIDs) that correspond to a set of input citation strings.
Required Parameters
db

Database to search. The only supported value is `pubmed'.
rettype

Retrieval type. The only supported value is `xml'.
bdata

Citation strings. Each input citation must be represented by a citation string in the following format:

journal_title|year|volume|first_page|author_name|your_key|

Multiple citation strings may be provided by separating the strings with a carriage return character (%0D). The your_key value is an arbitrary label provided by the user that may serve as a local identifier for the citation, and it will be included in the output. Be aware that all spaces must be replaced by `+' symbols and that citation strings should end with a final vertical bar `|'.

http://eutils.ncbi.nlm.nih.gov/entrez/eutils/ecitmatch.cgi?db=pubmed&retmode=xml&bdata=proc+natl+acad+sci+u+s+a|1991|88|3248|mann+bj|Art1|%0Dscience|1987|235|182|palmenberg+ac|Art2|
Go to:
Release Notes
ESearch &sort; JSON output format: February 14, 2014

    ESearch now provides a supported sort parameter
    EInfo, ESearch and ESummary now provide output data in JSON format

ECitMatch, EInfo Version 2.0, EFetch: August 9, 2013

    ECitMatch is a new E-utility that serves as an API to the PubMed batch citation matcher
    EInfo has an updated XML output that includes two new fields: <IsTruncatable> and <IsRangeable>
    EFetch now supports the BioProject database.

EFetch Version 2.0. Target release date: February 15, 2012

    EFetch now supports the following databases: biosample, biosystems and sra
    EFetch now has defined default values for &retmode and &rettype for all supported databases (please see Table 1 for all supported values of these parameters)
    EFetch no longer supports &retmode=html; requests containing &retmode=html will return data using the default &retmode value for the specified database (&db)
    EFetch requests including &rettype=docsum return XML data equivalent to ESummary output

Release of new Genome database: November 9, 2011

    Entrez Genome has been completely redesigned, and database records now correspond to a species rather than an individual chromosome sequence. Please see full details of the change at www.ncbi.nlm.nih.gov/About/news/17Nov2011.html
    Old Genome IDs are no longer valid. A file is available on the NCBI FTP site that maps old Genome IDs to Nucleotide GIs: ftp.ncbi.nih.gov/genomes/old_genomeID2nucGI
    EFetch no longer supports retrievals from Genome (db=genome).
    The ESummary XML for Genome has been recast to reflect the new data model.
    To view the new search fields and links supported for the new Genome database, please see http://eutils.ncbi.nlm.nih.gov/entrez/eutils/einfo.fcgi?db=genome 

ESummary Version 2.0. November 4, 2011

    ESummary now supports a new, alternative XML presentation for Entrez document summaries (DocSums). The new XML is unique to each Entrez database and generally contains more extensive data about the record than the original DocSum XML.
    There are no plans at present to discontinue the original DocSum XML, so developers can continue to use this presentation, which will remain the default.
    Version 2.0 XML is returned when &version=2.0 is included in the ESummary URL.

\subsection{European Nucleotide Archive - ENA}
La base de datos ENA ofrece varios servicios para la b˙squeda y descarga de 
registros programaticamente usando una interface REST. Adem·s da acceso FTP para
descargar sus datos.

De todos los servicios que ofrece, se discutir·n el servicio de recuperaciÛn de
datos, el de b˙squeda por texto y el acceso por FTP.

\subsubsection{RecuperaciÛn de datos - \textit{Data retrieval}}
La URL base de este servicio es '\url{http://www.ebi.ac.uk/ena/data/view/}'.
El uso m·s sencillo es la b˙squeda por el cÛdigo de identificaciÛn del
registro que se quiere recuperar. Por ejemplo, la b˙squeda
'\url{http://www.ebi.ac.uk/ena/data/view/KP718916}, recupera el registro
`KP718916'. Es posible, la b˙squeda por rango separando dos cÛdigos de
identificaciÛn con el sÌmbolo `-'.

Otra opciÛn, es la b˙squeda por nombre taxonÛmico. Este tipo de b˙squeda acepta
nombres comunes, nombres cientÌficos y n˙meros de identificaciÛn de taxa.
En este tipo de b˙squeda la URL base es
'\url{http://www.ebi.ac.uk/ena/data/view/Taxon:}' Por ejemplo, la b˙squedas que
se muestra debajo recuperan los registros asociados a una especie de laucha.
\begin{itemize}
  \item \url{http://www.ebi.ac.uk/ena/data/view/Taxon:calomys laucha}
  \item \url{http://www.ebi.ac.uk/ena/data/view/Taxon:small vesper mouse}
  \item \url{http://www.ebi.ac.uk/ena/data/view/Taxon:56211}
\end{itemize}

Adem·s, es posible obtener informaciÛn acerca de proyectos de secuenciaciÛn.
Para ello, se necesita el n˙mero de identificaciÛn del proyecto (project\_id) o
el cÛdigo de acceso del estudio (study\_accession). La URL para estas b˙squedas
se muestran debajo.
\begin{itemize}
  \item http://www.ebi.ac.uk/ena/data/view/study\_accession
  \item Ejemplo: \url{http://www.ebi.ac.uk/ena/data/view/PRJDB1793}
  \item http://www.ebi.ac.uk/ena/data/view/Project:project\_id
  \item Ejemplo: \url{http://www.ebi.ac.uk/ena/data/view/Project:254551}
\end{itemize}

El formato de descarga de los datos es por omisiÛn HTML, sin embargo, esta no 
es la mejor opciÛn si se desea hacer consulta program·ticas. ENA ofrece otros
formatos para descargar los datos. Para ello, se utiliza la opciÛn 
\emph{display} en el URL de b˙squeda. En la tabla \ref{tab:ena_fmt} se muestran
los formatos aceptados.

\begin{table}[!h]
\begin{center}
\small{}
\begin{tabular}{lp{5cm}p{7cm}}
OpciÛn & DescripciÛn & Ejemplo \\ \hline
html   & Valor por omisiÛn. Muestra los resultados en formato html. Disponible
para todos los tipos de datos &
\url{http://www.ebi.ac.uk/ena/data/view/KP718916&display=html} \\
xml    & Muestra los resultados en formato Xml. Disponible para todos los tipos
de datos & \url{http://www.ebi.ac.uk/ena/data/view/KP718916&display=xml} \\
text   & Muestra los datos en formato de texto plano. Disponible solo para
secuencias con anotaciÛn y ensambladas &
\url{http://www.ebi.ac.uk/ena/data/view/KP718916&display=text} \\
fasta  & Muestra los datos en formato fasta. Disponible solo para
secuencias con anotaciÛn, ensambladas y \textit{trace} & 
\url{http://www.ebi.ac.uk/ena/data/view/KP718916&display=fasta} \\ 
fastq  & Muestra los datos en formato fastq. Solo disponible para \textit{trace}
& \url{http://www.ebi.ac.uk/ena/data/view/TI1&display=fastq} \\
\end{tabular}
\normalsize{}
\end{center}
\caption{Formatos disponibles en ENA}
\label{tab:ena_fmt}
\end{table}

Adem·s de poder especificar el formato del contenido, se puede definir si se
desea que la descarga sea un archivo de texto plano o un archivo comprimido.
Para ello, se usa la opciÛn \emph{download}. Los valores permitidos son
\emph{txt} y \emph{gzip}.

Este sistema de descarga limita la cantidad de descargas a 100 000 registros por
consulta. Este lÌmite es para evitar la descarga no intencional de un gran
cantidad de datos. Si, es necesario descargar m·s registros se utiliza la 
opciÛn \emph{lentgh}, cuyo valor es el nuevo n˙mero de lÌmite que se desea. Otra
opciÛn importante es \emph{offset} que establece cual es el primer registro que
se descargar·. Esto es ˙til para hacer descargar fragmentadas.

\subsubsection{B˙squeda por texto}
La b˙squeda por texto en ENA usa la URL base \url{http://www.ebi.ac.uk/ena/data/warehouse/search}.
La b˙squeda se realiza en particiones de las bases de datos denominadas \emph{domain} 
y \emph{result}. La particiÛn \emph{domain} es de mayor nivel de abstracciÛn y 
divide el contenido seg˙n su naturaleza. La particiÛn \emph{result} es un 
refinamiento de \emph{domain} que divide la informaciÛn con alg˙n criterio
adiocional. Por ejemplo, el \emph{domain} `sequence' agrupa los registros que 
contiene informaciÛn de secuencias nucleotÌdicas. Para este \emph{domain} hay
dos \emph{result} disponibles: `sequence\_release'  y `sequence\_update'. El 
primero contiene todas las secuencias de la ˙ltima versiÛn publicada de la
base de datos y el segundo tiene las secuencias que se agregaron o actualizaron
desde esa ˙ltima publicaciÛn. Este ˙ltimo conjunto suele ser mucho m·s pequeÒo.
En la tabla \ref{tab:ena_domains_and_results} se muestran todos los \emph{domain} 
y \emph{result} disponibles. 

\begin{table}
\begin{center}
\begin{tabular}{llp{7cm}}
       Domain &             Result &                                                DescripciÛn \\ \hline
     assembly &           assembly &                                        Genomas ensamblados \\
     sequence &  sequence\_release &                        Secuencias de nucleÛtidos (release) \\
     sequence &   sequence\_update &                         Secuencias de nucleÛtidos (Update) \\
       coding &    coding\_release &                          Secuencias codificantes (Release) \\
       coding &     coding\_update &                           Secuencias codificantes (Update) \\
    noncoding & noncoding\_release &                       Secuencias no codificantes (Release) \\
    noncoding &  noncoding\_update &                        Secuencias no codificantes (Update) \\
         read &   read\_experiment &                  Lecturas crudas agrupadas por experimento \\
         read &          read\_run &                                            Lecturas crudas \\
         read &        read\_study &            Lecturas crudas agrupadas agrupadas por estudio \\
     analysis &           analysis & An·lisis de secuencias de nucleÛtidos a partir de lecturas \\
     analysis &    analysis\_study & An·lisis de secuencias de nucleÛtidos a partir de lecturas, agrupados por estudio \\
        trace &        read\_trace &                                        Trazas de capilares \\
        study &              study &                                                   Estudios \\
        taxon &              taxon &                                   ClasificaciÛn taxonÛmica \\
       sample &             sample &                                                   Muestras \\
environmental &      environmental &                                       Muestras ambientales \\
\end{tabular}
\end{center}
\caption{Particiones \emph{domain} y \emph{result} disponibles en ENA.}
\label{tab:ena_domains_and_results}
\end{table}

Las b˙squeda program·ticas por texto, adem·s del par·metro URL \emph{query},
requieren de los par·metros \emph{domain} y \emph{result}. El contenido del 
par·metro \emph{query} no es libre, es decir no es posible hacer una b˙squeda 
arbitraria de texto. El contenido del par·metro se construye por opciones de 
filtrado unidos por operadores lÛgicos (\emph{AND}, \emph{OR}, \emph{NOT})). 
Si se tienen varias opciones de filtrado, se pueden agrupar entre parÈntesis.

Las opciones de filtrado se aplican a los distintos campos que posee cada 
registro. Los campos disponibles varian en cada \emph{domain} y \emph{result}.
TambiÈn, cada campo de b˙squeda tienen valores de un tipo particular: \emph{booleano},
\emph{Vocabulario restringido}, \emph{fecha}, \emph{n˙merico}, \emph{textual},
\emph{geo-espacial} y \emph{taxonÛmico}. En la tabla \ref{tab:ena_search_fields}
se muestran algunos de los campos de b˙squeda m·s comunes para el \emph{domain=sequence}
y \emph{result=sequence\_release}. Estos campos se comparan con operadores de 
igualdad (`='), diferencia (`!='), mayor y menor (`>', `>=', `<=' y `<'), excepto
los campos de ubicaciÛn geogr·fica y taxonomÌa. La b˙squeda por estos campos
requiere el uso de funciones particulares. En la tabla \ref{tab:ena_search_loc_and_taxon}
se muestran algunas funciones que provee ENA para este tipo de b˙squeda.

\begin{table}[h!]
\begin{center}
\small
\begin{tabular}{lp{5cm}p{4cm}l}
FunciÛn   & DescripciÛn & Par·metros & Ejemplo \\ \hline
geo\_box1 & La ubicaciÛn se encuentra dentro de un cuadrado definido por la 
            esquina superior derecha y la esquina inferior izquierda                              
          & latitud suroeste, longitud suroeste, latitud noreste y longitud 
            noreste
          & geo\_box1(-20,10,20,50)\\
geo\_circ & La unicaciÛn se encuentra dentro de un circulo definido por el 
            centro y el radio 
          & Latitud, longitud y radio (km) 
          & geo\_circ(44,00,00) \\
tax\_eq    & Registros que coinciden con un identificador de taxonomÌa del NCBI 
          & Identificador de taxonomÌa de NCBI(numÈrico) 
          & tax\_eq(9606) \\
tax\_tree  & Registros que coinciden con un identificador de taxonomÌa del NCBI y
            todos sus descendientes. 
          & Identificador de taxonomÌa de NCBI(numÈrico) 
          & tax\_tree(2759) \\
\end{tabular}
\normalsize{}
\end{center}
\caption{Funciones para la b˙squeda por geolocalizaciÛn y por taxonomÌa}
\label{tab:ena_search_loc_and_taxon}
\end{table}

\begin{table}[h!]
\begin{center}
\small{}
\begin{tabular}{lp{3cm}p{7cm}}
            accession &                   Texto &                                         N˙mero de acceso \\ \hline
          base\_count &                  N˙mero &                                Longitud de la secuencias \\
           cell\_line &                   Texto &              lÌnea celular de donde se obtuvo la muestra \\
              country &                   Texto &        UbicaciÛn de la muestra: PaÌs, regiÛn y localidad \\
          description &                   Texto &                        Breve descripciÛn de la secuencia \\
environmental\_sample &                Booleano &  EspecifÌca si el material pertenece a muestra ambiental \\
             keywords &                   Texto &                  Palabras clave asociadas a la secuencia \\
        last\_updated &                   Fecha &                         Fecha de la ˙ltima actualizaciÛn \\
%             location &      Latitud y longitud &  UbicaciÛn geogr·fica de punto aislamiento de la muestra \\
            mol\_type & Vocabulario restringido &           Tipo molecular de la molÈcula \textit{in vivo} \\
\end{tabular}
\normalsize{}
\end{center}
\caption{Lista de algunos de los campos de b˙squeda ofrecidos por ENA.}
\label{tab:ena_search_fields}
\end{table}

La b˙squeda por texto ofrece las mismas opciones para manipular el formato de 
los datos a recibir que la b˙squeda por identificador que se discutiÛ en la 
primer secciÛn. Se usan los par·metros \emph{display}, \emph{download}, \emph{length}
y \emph{offset} ya vistos (ver la tabla \ref{tab:ena_fmt}). Este tipo de b˙squeda
ofrece una opciÛn m·s para el par·metro \emph{display}, el valor `report'. Esta 
opciÛn permite descargar algunos campos de los registros y no el registro completo.
Es ˙til para tener en un resumen de los registros que se desea descargar. Cuando
se usa esta opciÛn se requiere el uso del par·metro \emph{fields} para indicar 
los campos que se desea obtener. Debajo se muestran 
algunos de los campo m·s usuales para el \emph{domain} `sequence' y \emph{result}
`sequence\_release'.

\begin{multicols}{3}
\begin{itemize}
  \item accession
  \item base\_count
  \item cell\_line
  \item description
  \item environmental\_sample
  \item host 
  \item keywords 
  \item last\_updated 
  \item location 
  \item mol\_type 
  \item tax\_division
  \item tax\_id 
  \item scientific\_name
\end{itemize}
\end{multicols}

La referencia completa de los campos de b˙squeda y los campos disponibles para 
los reportes se puede consultar en \url{http://www.ebi.ac.uk/ena/data/warehouse/usage}.

\subsubsection{Formato del registro}
\scriptsize{}
\begin{verbatim}
ID   AJ131281; SV 1; linear; genomic RNA; STD; VRL; 348 BP.
XX
AC   AJ131281;
XX
DT   03-DEC-1998 (Rel. 57, Created)
DT   10-DEC-1999 (Rel. 62, Last updated, Version 2)
XX
DE   Lymphocytic choriomeningitis virus z gene, partial, strain MX
XX
KW   RING finger protein; z gene.
XX
OS   Lymphocytic choriomeningitis mammarenavirus
OC   Viruses; ssRNA viruses; ssRNA negative-strand viruses; Arenaviridae;
OC   Mammarenavirus.
XX
RN   [1]
RP   1-348
RA   Pastorek J.;
RT   ;
RL   Submitted (02-DEC-1998) to the INSDC.
RL   Pastorek J., Molecular Biology, Institute of Virology, Slovak Academy of
RL   Sciences, Dubravska cesta 9, 842 46, SLOVAK REPUBLIC.
...
XX
DR   MD5; 520554560ea4a63dc1fe3f175b37a319.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..348
FT                   /organism="Lymphocytic choriomeningitis mammarenavirus"
FT                   /lab_host="HeLa cells"
FT                   /strain="MX"
FT                   /mol_type="genomic RNA"
FT                   /db_xref="taxon:11623"
FT   CDS             <70..343
FT                   /codon_start=2
FT                   /gene="z"
FT                   /product="ring finger protein"
...
FT                   /experiment="experimental evidence, no additional details
FT                   recorded"
FT                   /protein_id="CAA10342.1"
FT                   /translation="MGQGKSKEKKDTNTGDRAEILPDTTYLGPLNCKSCWQKFDSLVRC
FT                   HDHYLCRHCLNLLLSVSDRCPLCKCPLPTKLKISTAPSPPPPYEE"
FT   mat_peptide     <70..338
FT                   /gene="z"
FT                   /product="ring finger protein"
XX
SQ   Sequence 348 BP; 95 A; 100 C; 77 G; 76 T; 0 other;
     cgtttagttg cgctgtttgg ttgaacagcc ttttcctgtg agagtacaga gacaaaccta        60
...
     gatatcaaca gccccaagcc caccacctcc ctacgaagag taacaccg                    348
//
\end{verbatim}
\normalsize{}
\subsection{DNA Data Bank of Japan - DDBJ}
La base de datos DDBJ ofrece, adem·s de la interfaz web, el acceso por FTP y
una API web llamada WABI (Web API for Biology). Estos dos son los recursos que
permiten el acceso program·tico a esta base de datos.

\subsubsection{Formato del registro}
\scriptsize{}
\begin{verbatim} 
LOCUS       AJ131281                 348 bp    RNA     linear   VRL 10-DEC-1999
DEFINITION  Lymphocytic choriomeningitis virus z gene, partial, strain MX.
ACCESSION   AJ131281
VERSION     AJ131281.1  GI:3970751
KEYWORDS    RING finger protein; z gene.
SOURCE      Lymphocytic choriomeningitis mammarenavirus (LCMV)
  ORGANISM  Lymphocytic choriomeningitis mammarenavirus
            Viruses; ssRNA viruses; ssRNA negative-strand viruses;
            Arenaviridae; Mammarenavirus.
REFERENCE   1
  AUTHORS   Gibadulinova,A., Zelnik,V., Reiserova,L., Zavodska,E.,
            Zatovicova,M., Ciampor,F., Pastorekova,S. and Pastorek,J.
  TITLE     Sequence and characterisation of the Z gene encoding ring finger
            protein of the lymphocytic choriomeningitis virus MX strain
  JOURNAL   Acta Virol. 42 (6), 369-374 (1998)
   PUBMED   10358742
REFERENCE   2  (bases 1 to 348)
  AUTHORS   Pastorek,J.
  TITLE     Direct Submission
  JOURNAL   Submitted (02-DEC-1998) Pastorek J., Molecular Biology, Institute
            of Virology, Slovak Academy of Sciences, Dubravska cesta 9, 842 46,
            SLOVAK REPUBLIC
FEATURES             Location/Qualifiers
     source          1..348
                     /organism="Lymphocytic choriomeningitis mammarenavirus"
                     /mol_type="genomic RNA"
                     /strain="MX"
                     /db_xref="taxon:11623"
                     /lab_host="HeLa cells"
     gene            <70..343
                     /gene="z"
     CDS             <70..343
                     /gene="z"
                     /experiment="experimental evidence, no additional details
                     recorded"
                     /codon_start=2
                     /product="ring finger protein"
                     /protein_id="CAA10342.1"
                     /db_xref="GI:3970752"
                     /translation="MGQGKSKEKKDTNTGDRAEILPDTTYLGPLNCKSCWQKFDSLVR
                     CHDHYLCRHCLNLLLSVSDRCPLCKCPLPTKLKISTAPSPPPPYEE"
     mat_peptide     <70..338
                     /gene="z"
                     /product="ring finger protein"
ORIGIN      
        1 cgtttagttg cgctgtttgg ttgaacagcc ttttcctgtg agagtacaga gacaaaccta
 ...
      301 gatatcaaca gccccaagcc caccacctcc ctacgaagag taacaccg
//
\end{verbatim}
\normalsize{}
\subsubsection{Acceso FTP}
DDBJ permite descargas varias bases de datos. Nos enfocaremos en \emph{DDBJ}
(nucleotidos), \emph{DAD} (amino·cidos) y \emph{wgs} (secuenciaciÛn de genomas
completos). 
La URL base para ellas son:
\begin{description}
  \item[ddbj] \url{ftp://ftp.ddbj.nig.ac.jp/ddbj_database/ddbj}
  \item[ddbj - release notes]
  \url{ftp://ftp.ddbj.nig.ac.jp/ddbj_database/release_note_archive/ddbj}
  \item[dad]  \url{ftp://ftp.ddbj.nig.ac.jp/ddbj_database/dad}
  \item[dad - release notes] 
  \url{ftp://ftp.ddbj.nig.ac.jp/ddbj_database/release_note_archive/dad}
  \item[wgs]  \url{ftp://ftp.ddbj.nig.ac.jp/ddbj_database/wgs}
  \item[wgs - Lista de organismos]
  \url{ftp://ftp.ddbj.nig.ac.jp/ddbj_database/wgs/WGS_ORGANISM_LIST.txt}
\end{description}

La informaciÛn en las bases de datos \emph{ddbj} y \emph{dad} est· organizada
en diferentes categorÌas, seg˙n el origen de las secuencias. Los archivos de 
datos que pueden ser descargados se corresponden con estas categorÌas y siguen
una nomenclatura racional. En la tabla \ref{ddbj_ftp_categories} se detallan las
categorÌas. El nombre de cada archivo est· formado por un prefijo que 
corresponde con alguna de las categorÌas, despuÈs un n˙mero de orden creciente y
finalmente la extensiÛn. El formato de los archivos es texto plano, comprimido.
Para las bases de datos \emph{ddbj} y \emph{dad}, posible descargar archivos en 
formato \texttt{fasta} y \texttt{XML}. Estos se encuentran en las
subcarpetas \texttt{fasta/} y \texttt{xml/insdxml/v1.4} respectivamente.

\begin{table}[h!tp]
\begin{center}
\begin{tabular}{llp{11cm}}
CategorÌa  & Prefijo & DescripciÛn \\ \hline
HUM        & ddbjhum & Humano \\
PRI        & ddbjpri & Primates (no humano) \\
ROD        & ddbjrod & Roedores \\
MAM        & ddbjmam & Mamiferos (no primates, ni roedores) \\
VRT        & ddbjvrt & Vertebrados (no mamÌferos) \\
INV        & ddbjinv & Invertebrados \\
PLN        & ddbjpln & Plantas, hongos, plastidos y otros eukariotas \\
BCT        & ddbjbct & Bacterias y archaeas \\
VRL        & ddbjvrl & Viruses \\
PHG        & ddbjphg & Bacteriofagos \\
ENV        & ddbjenv & Secuencias de muestras ambientales \\
SYN        & ddbjsyn & Secuencias de cosntrucciones sintÈticas \\
EST        & ddbjest & \textit{Expressed sequence tags}. Secuencias cortas
                       de cDNA \\
GSS        & ddbjgss & \textit{Genome survey sequences}. Secuencias
                       genÛmicas cortas \\
TSA        & ddbjtsa & Ensamblados de transcriptÛmica (\textit{Transcriptome
                       shhotgun assemblies}) \\
HTC        & ddbjhtc & Secuencias de cDNA por mÈtodos de alto rendimiento
                       (\emph{high throughput} .
                       Esta categorÌa incluye secuencias de cDNA no terminadas.
                       Cuando la secuenciaciÛn se completa se mueve a la 
                       categorÌa taxonÛmica que le corresponde \\
HTG        & ddbjhtg & Secuencias genÛmicas por mÈtodos de alto rendimiento
                       (\emph{high throughput}) \\
STS        & ddbjsts & \emph{Sequence tagged sites} \\
PAT        & ddbjpat & Secuencias relacionadas con aplicaciones patentadas \\
UNA        & ddbjuna & Secuencias no anotadas. \\
CON        & ddbjcon & Rearmados en proyectos de secuenciaciÛn (Contigs) \\
\end{tabular}
\end{center}
\caption{CategorÌas de las bases de datos de DDBJ}
\label{ddbj_ftp_categories}
\end{table}


\subsubsection{Acceso por Web API}
DDBJ ofrece API para tres servicios diferentes. El servicio WABI BLAST, que 
permite la b˙squeda usando Blast. El servicio 'getentry WebAPI' que permite la
recuperar registros de la base de dato por n˙mero de acceso y el servicio ARSA 
WebAPI, que permite la b˙squeda de registros por su contenido textual (no de 
secuencia. Los tres servicios usan una arquitectura REST 
(\textit{Representational state transfer}) sobre http para las consultas. 



\subsubsection{Getentry WebAPI}
El servicio getentry permite recuperar registro de una de las bases de data de 
DDBJ por su n˙mero de acceso. Las consultas se pueden hacer mediante el mÈtodo
HTML GET o usando una URL sem·ntica. En ambos casos, la URL base es la misma:
'\url{http://getentry.ddbj.nig.ac.jp/getentry}'. La sÌntaxis para especificar 
una consulta usando el mÈtodo GET es '\url{http://getentry.ddbj.nig.ac.jp/getentry?database=nombre_de_la_base_de_datos&accession_number=n˙mero_de_acceso}' \\
por ejemplo, la consulta '\url{http://getentry.ddbj.nig.ac.jp/getentry?database=ddbj&accession_number=KP177965}' \\
recupera el registro con el n˙mero de acceso KP177965. 
La sÌntaxis para hacer una consulta mediante una URL sem·ntica
La misma consulta, usando
una URL sem·ntica es
'\url{http://getentry.ddbj.nig.ac.jp/getentry/nombre_de_la_base_de_datos/n˙mero_de_acceso/revision},
donde 'revisiÛn' es opcional. Los registros pueden ser actualizados en el 
tiempo, cada nueva versiÛn de un registro se carateriza por la fecha y hora de
la actualizaciÛn. Por ejemplo, la
'\url{http://getentry.ddbj.nig.ac.jp/getentry/ddbj/AY646354/2010-02-26+14:07:01}'
\\ recupera la revisiÛn 2010-02-26 14:07:01 del registro. Para saber que 
revisiones est·n disponibles para un registro particular se puese hacer una
consulta a la '\url{http://getentry.ddbj.nig.ac.jp/gethistory}', con un formato
similar a la de 'getentry'.

Los par·metros disponibles para las consultas son el \emph{n˙mero de acceso},
\emph{base de datos}, \emph{revisiÛn}, \emph{formato}, \emph{tipo de archivo},
\emph{mostrar eliminados} y \emph{limite}.


Para especificar el n˙mero se acceso se utiliza la palabla clave
\emph{accession\_number}. Se puede indicar el n˙mero de acceso solo (ej.
AY646354), o el n˙mero de versiÛn (AY646354.1). Se puede especificar varios 
n˙mero de acceso en un sola consulta, separ·ndolos con una coma. TambiÈn se 
puede indicar un rango de n˙meros de acceso, conectando dos n˙mero de accesos
con un sÌmbolo '-'. Este par·metros es el ˙nico obligatorio.

Para especificar una base de datos se usa la palabra clave \emph{database}.
Las b·ses de datos disponibles se muestran en la tabla \ref{tab:getentry_dbs}

\begin{table}
\begin{center}
\begin{tabular}{llp{7cm}}
Tipo    & Base de datos & DescripciÛn\\ \hline
DNA     & na            & Busca en todas las bases de datos dispoibles. Es la
                          opciÛn usada por defecto. \\
        & ddbj          & Base de datos propioa de DDBJ. \\
        & wgs           & Base de datos de proyectos de secuenciaciÛn de
                          genomas completos (Whole genome sequences). \\
        & mga           & SecuenciaciÛn masiva para anotaciÛn genÛmica (Mass
                          sequence for Genome Annotation. \\ \hline
Protein & aa            & Busca en todas las bases de datos disponibles. \\
        & uniprot       & Base de datos Uniprot.\\
        & dad           & Secuencias de amino·cidos de DDBJ. \\
        & patent\_aa    & Secuencias derivadas de patentes. 

El cÛdigo de revisiÛn se especifica con la palabra clave \emph{revision} y 
consiste en la fecha y la hora con el formato \ttfamily{yyyy-MM-dd hh:mm:ss} o
\ttfamily{yyyy-MM-dd hh:mm:ss release}.

\end{tabular}
\end{center}
\caption{Bases de datos disponibles en \emph{gententry}}
\label{tab:getentry_dbs}
\end{table}

Para especificar el formato de retorno de los datos se utiliza la palabra clave
\emph{format}. En la tabla \ref{tab:getentry_fmts} se muestran los posibles 
valores que puede tomar.

\begin{table}
\begin{center}
\begin{tabular}{ll}
Formato  & DescripciÛn             \\ \hline
flatfile & Formato de texto plano \\
xml      & Formato INSDSeq-XML     \\
fasta    & Secuencias de DNA o proteÌna en formato fasta \\
trans    & Secuencia codificante traducida a proteÌnas (fasta) \\
cds      & Secuencia codificante en nucleÛtidos (fasta)
\end{tabular}
\end{center}
\caption{Formatos de retorno de \emph{getentry}.}
\label{tab:getentry_fmts}
\end{table}

Para especificar que tipo de archivo en que se quiere recuperar los datos se 
usa la palabra clave \emph{filetype}. Los valores posibles son \emph{html}, 
\emph{text} y \emph{gz} (texto plano comprimido).

Es posible recuperar informaciÛn de registro viejos y que fueron eliminados de
las versiones m·s recientes de las bases de datos. Para ello, se debe usar la
palabra clave \emph{show\_suppressed}. Los valores posibles son \emph{true} y 
\emph{false}. El valor por defecto es \emph{false}.

Para limitar la cantidad de resultados que se desea recuperar se utiliza el
par·metros \emph{limit}. Cuyo valor es el n˙mero m·ximo de registro que se 
puede recuperar. El valor por defecto es 10. Si se especifica 0, se repueran
todos los registros disponibles.


\subsubsection{ARSA WebAPI.} Est· diseÒada para ser usada con un navegador web. Los
navegadores compatibles son Firefox y Chrome, IE versiones 9 y 10; y Safari
version 5 y posteriores.

Arsa recupera informaciÛn de las siguientes bases de datos:
\begin{itemize}
  \item DDBJ Release
  \item DDBJ New (daily updates)
  \item Amino Acid Patent Sequence data submitted from JPO
  \item Amino Acid Patent Sequence data submitted from KIPO.
\end{itemize}

La URL base para la b˙squeda es '\url{http://ddbj.nig.ac.jp/arsa/search}'. Los 
resultados pueden ser recuperados en idioma inglÈs o japonÈs. Para definir el
idioma es necesario agregar a la direcciÛn base el el capo 'lang' con valor 
'en'. De esta forma la URL se convierte en: 
\url{http://ddbj.nig.ac.jp/arsa/search?lang=en}. El formato de respuesta es HTML,
porque est· pensado para ser mostrado en un navegador como una p·gina web.
Si se desea, obtener esta informaciÛn en un formato que pueda ser f·cilmente 
manipulado se tienen dos opciones. Ambas implican usar una URL base diferente. 
Una de las opciones es descargar el contenido de todos los registros en formato
plano, para ello, hay que cambiar a la URL '\url{http://ddbj.nig.ac.jp/arsa/downloadAll?lang=en}'.
La otra opciÛn es recuperar los identificadores de cada registro y luego 
descargar cada uno de ellos, invididualmente con el servicio 'getentry', para 
ello, se debe cambiar la URL base a '\url{http://ddbj.nig.ac.jp/arsa/searchAllIds?lang=en}'. 

Hay dos condiciones de b˙squeda, la b˙squeda r·pida (\textit{Quick Search}) y la
b˙squeda avanzada (\textit{Advanced Search}). La condiciÛn de b˙squeda se define
en el campo 'cond', sus posibles valores son: 'quick\_search' y 'advanced\_search'.
Sin embargo, en esta guÌa se trabajar· solamente con la condiciÛn de b˙squeda 
simple. El campo 'query' define el texto que se desea buscar. Existen varios opciones
para refinar la b˙squeda. Los valores que corresponden a valores booleanos
('AND', 'OR', 'NOT') deben ser escritos en may˙sculas. Para separar dos o m·s
palabras, en lugar de usar un espacio, se debe usar el signo '+'. 

Las opciones de la b˙squeda r·pida son las siguientes:
\begin{description}
  \item[Coincidencia por conjunciÛn] B˙squedas que contienen todas las palabras clave 
  dadas. Se debe usar el campo 'operator' con el valor 'AND'. Por ejemplo, la
  b˙squeda '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=brain+phosphatidylethanolamine&operator=AND}'\\  
  recupera todos los registros que contienen las palabras 'brain' y
  'phosphatidylethanolamine', aunque se encuentren en lugares diferentes en el
  registro. El primer resultado de la b˙squeda del ejemplo es un registro que
  contiene la palabra 'brain' en la definiciÛn del registro (tÌtulo principal)
  y la palabra 'phosphatidylethanolamine' en el tÌtulo de una de las referencias
  bibliogr·ficas que aparecen en ese registro.
  \item[Coincidencia por disyunciÛn] B˙squedas que contienen al menos una de 
  las palabras clave dadas. Se indican en el campo 'operator' y con el valor 
  'OR'. Por ejemplo, la b˙squeda '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=tinea+versicolor&operator=OR}'\\
  recupera registros de 'Tinea versicolor', 'Trametes versicolor' y 'Tinea pellionella'.
  \item[NegaciÛn] B˙squedas que contienen una o m·s palabras clave y que carecen
  de otras que son dadas explicitamente. Se indican con la palabra clave 'NOT'
  dentro del campo 'query', la palabra clave que se encuentra  a su derecha es 
  la que debe estar ausente. Por ejemplo, la b˙squeda
  '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=homo+NOT+sapiens&operator=AND}'\\
  recupera los registros que contiene la palabros 'homo' pero no 'sapiens'.
  \item[Coincidencia parcial] B˙squedas que contienen parte de una palabra. Se
  indica en el campo 'query'. Al final del tÈrmino buscado se agrega el
  sÌmbolo '*'. Por ejemplo, la b˙squeda '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=candid*}'\\
  recupera registros de '\textit{Candida dubliniensis}', '\textit{Candidatus Phytoplasma}',
  'Junin virus strain candid1', entre otros.
  \item[Coincidencia de frases] B˙squedas que contienen varios tÈrminos en el orden
  dado. Se indican encerrando entre comillas los tÈrminos. Dependiendo de la 
  herramienta que se utilice, puede ser necesario reemplazar
  las comillas por '\%22'. Por ejemplo, la b˙squeda 
  '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=%22mouse+brain%22&operator=AND
  }'\\
  recupera los registro que contengan la frase 'mouse brain'.
  \item[B˙squedas en campos] B˙squedas que deben coincidir en un campo 
  particular del registro. Se indican nombrando el campo en el que se debe 
  hacer la b˙squeda antes de la palabra clave buscada y separados por el 
  sÌmbolo ':' (o su cÛdigo de escape '\%3A'). Por ejemplo, la b˙squeda
  '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=df:``homo+sapiens"&operator=AND}' \\
  recupera registros que tienen la frase 'homo sapiens' en su definiciÛn (tÌtulo principal).
  \item[B˙squeda por expresiones regulares] Alg˙nas b˙squedas en campos permiten
  usar expresiones regulares. Deben encerrarse con el sÌmbolo '/' (o '\%2F'). 
  En la tabla \ref{DDBJ_ARSA_regex} se muestran las expresiones regulares 
  permitidas.
  \item[B˙squeda por rango de valores] Algunas b˙squedas en campos permiten 
  indicar un rango de valores para la coincidencia. Deben encerrarse entre 
  corchetes. La palabra clave 'TO' debe estar presente entre el valor
  mÌnimo y el m·ximo del rango. Se puede usar el sÌmbolo '*' para indicar un 
  valor de inicio o final del rango desconocido. Por ejemplo, la b˙squeda 
  '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=pa:[GL000000+TO+GL000010]&operator=AND}' \\
  recupera los registros existentes cuyo n˙mero de acceso primario sea 
  'GL000001', 'GL000002',  \ldots y 'GL000010'. Otro ejemplo,  '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=sl:[0+TO+100]&operator=AND}' \\
  recupera todos los registros cuya longitud de secuencia sea 100 o menos.
\end{description}


\begin{table}
\small{}
\begin{center}
\begin{tabular}{lrp{7cm}}
Nombre del campo        & Abr. & Description  \\ \hline
PrimaryAccessionNumber  & pa   & Primer n˙mero de acceso en ACCESSION \\
AccessionNumber         & an   & N˙meros de acceso en ACCESSION \\
Division                & dv   & Division en LOCUS (ej. VRL=virus) \\
SequenceLength          & sl   & Longitud de secuencia en LOCUS \\
MolecularType           & mt   & Tipo de molecula in LOCUS (ej. RNA) \\
MolecularForm           & mf   & Estructura molecular (ej. linear) \\
Date                    & dt   & ⁄ltima fecha de publicaciÛn en LOCUS  \\
Definition              & df   & Texto en DEFINITION \\
Comment                 & cm   & Texto en COMMENT \\
Keyword                 & kw   & Texto en KEYWORDS \\
Organism                & og   & Texto en el organismo de ORGANISM \\
Lineage                 & ln   & Texto en el linaje de ORGANISM \\
ReferenceAuthor         & ra   & Texto en AUTHORS de REFERENCE \\
ReferenceTitle          & rt   & Texto en TITLE de REFERENCE \\
ReferenceJournal        & rj   & Texto en JOURNAL de REFERENCE \\
ReferencePubmedID       & rp   & Texto en PUBMED de REFERENCE \\
Feature                 & fe   & Texto en FEATURES \\
\end{tabular}
\end{center}
\normalsize{}
\caption{Campos de b˙squeda disponibles en DDBJ}
\label{tab:DDBJ_fields}
\end{table}

\begin{table}
\begin{center}
\small
\begin{tabular}{p{4cm}lp{8cm}}
Significado                             & Simbolo   & Ejemplo \\ \hline
Cualquier caracter individual           & .         & /ho.o/ coincide con 'homo' y con 'Rhodococcus'. \\
Cualquier caracter cero o m·s veces     & *         & /XX1*/ coincide con 'XX', 'XX1', 'XX11', etc \\
Cualquier caracter cero o una vez       & ?         & /XY?123/ coincide con 'XY123' and 'X123' \\
Cualquier caracter una o m·s veces      & +         & /XY1*/ coincide con 'XY1', 'XY11',etc, pero no con 'XY' \\
Cualquier caracter en un grupo          & [abc]     & /ho[dm]o/ coincide con 'homo' y con 'Rhodococcus', pero no con 'Achoropsyche' \\
Cualquier caracter excepto algunos      & [\^{}abc] & /ho[\^{}d]o/ coincide con 'homo'y con 'Achoropsyche', pero no con 'Rhodococcus' \\
Cualquier caracter en un rango          & [a-z0-9]  & /XY[1-9]00/ coincide con 'XY100', 'XY200', etc, 'XYA00', 'XYB00', etc \\
Cualquier caracter exactamente n vences & \{n\}     & /XY1{2}/ coincide con 'XY11' pero no con 'XY0' o 'XY000' \\
Cualquier caracter n o m·s veces        & \{n,\}    & /XY1{2,}/ coincide con 'XY11' y 'XY111', pero no con 'XY1' \\
Cualquier caracter n hasta m veces      & \{n,m\}   & /XY1{2,3}/ coincide con 'XY11' y con 'XY111', pero no con 'XY1' o 'XY1111' \\
\end{tabular}
\normalsize{}
\end{center}
\caption{tablecaption}
\label{tablelabel}
\end{table}

\subsubsection{WABI BLAST}
DDBJ ofrece una API web, e interface para hacer consultas con Java, perl y Ruby.
Sin embargo, la documentaciÛn solo est· disponible en japonÈs, por ello lo 
dejaremos fuera de la discusiÛn. Si se desea ver en detalle la documentaciÛn 
disponible, visite la p·gina
'\url{http://www.ddbj.nig.ac.jp/search/help/wabi/wabi_blast_example-j.html}'.



% Las opciones de la b˙squeda r·pida son, en principio, similares a las de la 
% b˙squeda r·pida, sin embargo se permite un mayor flexibilidad para construir
% criterios de selecciÛn m·s complejos. Se permite hacer b˙squedas con operadores
% de disyunciÛn ('OR'), conjunciÛn ('AND') y negaciÛn ('NOT') simult·neamente en 
% un mismo campo. TambiÈn, se permite agrupar criterios entre parÈntesis. En esta
% condiciÛn de b˙squeda, el campo 'query' en la \textit{url} de la b˙squeda ya no
% se utiliza, en cambio cada campo de b˙squeda tiene uno que le corresponde en 
% la \textit{url} de b˙squeda. Por ejemplo, la b˙squeda r·pida 
% '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=quick_search&query=df:homo&operator=AND}'
% se convierte en '\url{http://ddbj.nig.ac.jp/arsa/search?lang=en&cond=advanced_search&df=homo&op=AND}'.
% Por ejemplo la b˙squeda
% '\url{http://ddbj.nig.ac.jp/arsa/search?lang=&cond=advanced_search&df=(virus+NOT+Junin)&ln=Arenaviridae&op=AND}'
% recupera los registros que en su linaje posea la palabra 'Arenaviridae' y que en
% la deficiÛn posea la palabra 'virus' y que no posea la palabra 'Junin'. Otro 
% ejemplo m·s, la b˙squeda 
% '\url{http://ddbj.nig.ac.jp/arsa/search?lang=&cond=advanced_search&df=((virus+AND+Junin)+NOT+segment)&op=AND}'
% recupera los registros que poseen las palabras 'virus' y 'junÌn' en la definiciÛn 
% y que no contengan la palabra 'segment'.
% Otro ejemplo, la b˙squeda 
% '\url{http://ddbj.nig.ac.jp/arsa/search?lang=&cond=advanced_search&df=((virus+AND+Junin)+OR+(virus+AND+Tacaribe))&op=AND}'
% recupera los que poseen los registros que contienen las palabras 'Junin' o 
% 'Tacaribe', pero no 'virus'.
% Todos los campos de b˙squeda disponibles est·n en la tabla
% \ref{tab:DDBJ_fields}.  

 


% 
% 



\end{document}
